File: nattybear/level1/최대공약수와-최소공배수/solution.js
function solution(n, m) {
  const g = gcd(n, m)
  const l = lcm(n, m)
  return [g, l]
}

function gcd(n, m) {
  if (m == 0)
    return n
  else
    return gcd(m, n % m)
}

function lcm(n, m) {
  return n * m / gcd(n, m)
}

File: nattybear/level1/최대공약수와-최소공배수/Solution.java
class Solution {
  public int[] solution(int n, int m) {
    int g = gcd(n, m);
    int l = lcm(n, m);
    return new int[] {g, l};
  }

  static int gcd(int n, int m) {
    if (m == 0)
      return n;
    else
      return gcd(m, n % m);
  }

  static int lcm(int n, int m) {
    return n * m / gcd(n, m);
  }
}
